### PR TITLE
DropOut and FastCell Support Added

### PR DESCRIPTION
**Pull request by SachinG007**

### CODE REVIEW

File: pytorch/examples/SRNN/SRNN_Example.py [hunk 1]
```diff
@@ -51,8 +51,24 @@
...
+
+Example OPTIONAL args for FastGRNNCell
+cellArgs = {'gate_non_linearity':"sigmoid",'update_non_linearity':"tanh",
+				'wRank':None, 'uRank':None,'zetaInit':1.0, 'nuInit':-4.0, 
+				'batch_first':False}
+
+'''
+cellArgs = {}
+
+srnn2 = SRNN2(numInput, numClasses, hiddenDim0, hiddenDim1, cellType,
```
Comment (metastableB) [inline pytorch/examples/SRNN/SRNN_Example.py]: Align arguments with braces. Install some PEP8 based linter on your editer (Sublime?) to automate this

File: pytorch/examples/SRNN/SRNN_Example.py [hunk 2]
```diff
@@ -51,8 +51,24 @@
...
+dropoutProbability_l0 = 0.2
+dropoutProbability_l1 = 0.2
 
-srnn2 = SRNN2(numInput, numClasses, hiddenDim0, hiddenDim1, cellType).to(device)
+print(cellType)
+'''
+cellArgs (optional) will be passed to the respective cell
+
+Example OPTIONAL args for FastGRNNCell
+cellArgs = {'gate_non_linearity':"sigmoid",'update_non_linearity':"tanh",
```
Comment (metastableB) [inline pytorch/examples/SRNN/SRNN_Example.py]: this is not how  dicts are formatted. Please use PEP8 linter

File: pytorch/pytorch_edgeml/graph/rnn.py [hunk 3]
```diff
@@ -1050,29 +1049,50 @@ def forward(self, input, hiddenState=None, cellState=None):
...
         '''
         super(SRNN2, self).__init__()
+
         # Create two RNN Cells
         self.inputDim = inputDim
         self.hiddenDim0 = hiddenDim0
         self.hiddenDim1 = hiddenDim1
         self.outputDim = outputDim
-        supportedCells = ['LSTM']
+        self.dropoutProbability_l0 = 0
```
Comment (metastableB) [inline pytorch/pytorch_edgeml/graph/rnn.py]: This should be `None` and not 0

File: pytorch/pytorch_edgeml/graph/rnn.py [hunk 4]
```diff
@@ -1117,12 +1137,25 @@ def forward(self, x, brickSize):
         x_bricks = torch.reshape(x_bricks, [oldShape[0], oldShape[1] * oldShape[2], oldShape[3]])
         # x bricks: [brickSize, numBricks * batchSize, featureDim]
         # x_bricks = torch.Tensor(x_bricks)
-        hidd0, out0 = self.rnn0(x_bricks)
+        dropoutLayer_0 = nn.Dropout(p = self.dropoutProbability_l0)
```
Comment (metastableB) [inline pytorch/pytorch_edgeml/graph/rnn.py]: Nope.

Lets change the way you are dealing with dropout to make it crisp and short.


```
# at the top of __init__, just assign this to self. variables
self.dropoutProbability_l0 = dropoutProbability_l0
self.dropoutProbability_l1 = dropoutProbability_l1
# . . .
# rest of the code till you define the cells
# . . .

self.dropoutLayer0 = None
self.dropoutLayer1 = None
hidd0 = rnn0(whatever)
if self.dropoutProbability0 is not None:
    self.dropoutLayer0 = nn.Dropout() # define this
    hidd0 = self.dropoutLayer0(hidd0, . . .)

hidd1 = self.rnn1(hidd0, . . .)
if self.dropoutProbability1 is not None:
    self.dropoutLayer1 = nnDropout()
    hidd1 = self.dropout(. . .)
# . . .
# Continue to FC layer
# . . .
    
```
